rules-reviewed/camel3/camel2/component-changes.windup.groovy

```diff
@@ -0,0 +1,308 @@
+                }
+            }
+        })
+        .withId("component-changes-00002")
+
+        .addRule()
+        .when(FileContent.matches("{*}.{method}{*}").inFileNamed("{*}.java"))
+        .perform(createHint("Mock: {method} has been removed",
+                "`{method}` has been removed from the mock component's assertion api.",
+                "out_message_removed_from_simple_language_and_mock_component", true))
+        .where("method").matches("(outBody|outHeaders)")
+        .withId("component-changes-00003")
+
+        .addRule()
+        .when(JavaClass.references("org.apache.camel.OutHeaders").at(TypeReferenceLocation.ANNOTATION, TypeReferenceLocation.IMPORT))
+
+        .perform(createHint("@OutHeaders annotation has been removed",
+                "`@OutHeaders` annotation has been removed. Use `@Headers` instead",
+                "out_message_removed_from_simple_language_and_mock_component", true))
+        .withId("component-changes-00004")
+
+        .addRule()
+        .when(XmlFile.matchesXpath("//*/c:simple[text()=windup:matches(self::node(), '{*}out.{params}{*}')]")
+                .namespace("c", "http://camel.apache.org/schema/spring"))
+        .perform(createSimpleLanguageRemovedHint("out.{params}"))
+        .where("params").matches("(body|header)")
+        .withId("component-changes-00005")
+
+        .addRule()
+        .when(XmlFile.matchesXpath("//*/b:simple[text()=windup:matches(self::node(), '{*}out.{params}{*}')]")
+                .namespace("b", "http://camel.apache.org/schema/blueprint"))
+        .perform(createSimpleLanguageRemovedHint("out.{params}"))
+        .where("params").matches("(body|header)")
+        .withId("component-changes-00006")
+
+        .addRule()
+        .when(FileContent.matches("{*}simple({*}out.{params}{*})").inFileNamed("{*}.java"))
+        .perform(createSimpleLanguageRemovedHint("out.{params}"))
+        .where("params").matches("(body|header)")
+        .withId("component-changes-00007")
+
+        .addRule()
+        .when(Or.any(
+                XmlFile.matchesXpath("//*/c:simple[text()=windup:matches(self::node(), '{*}property{*}')]")
+                        .namespace("c", "http://camel.apache.org/schema/spring"),
+                XmlFile.matchesXpath("//*/b:simple[text()=windup:matches(self::node(), '{*}property{*}')]")
+                        .namespace("b", "http://camel.apache.org/schema/blueprint"),
+                FileContent.matches("{*}simple({*}property{*})").inFileNamed("{*}.java"))
+        )
+        .perform(createHint("Simple language: property function has been removed",
+                "`property` function has been removed from simple language. Use `exchangeProperty` instead.",
+                "languages", true))
+        .withId("component-changes-00008")
+
+        .addRule()
+        .when(Or.any(JavaClass.references("org.apache.camel.component.hl7.HL7.terser").at(TypeReferenceLocation.IMPORT).as("terser"),
```

Comment:
Looks weird. 
The second condition works with the first one results, so I dont see the need for the second condition as if the first one passes, global result will be True no matter the result of the second condition.

Reply:
that's actually valid point! I believe I wanted to have "And" there. I have to revisit